src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableResolver.cs

```diff
@@ -0,0 +1,190 @@
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Basic implementation of INameResolver around a <see cref="PowerFxConfig"/> object. 
+    /// This aides in binding and intellisense. 
+    /// Host can override Lookup to provide additional symbols to the expression. 
+    /// </summary>
+    internal class SymbolTableResolver : INameResolver, IGlobalSymbolNameResolver
+    {
+        // $$$ Also needs IGlobalSymbolNameResolver? 
+
+        private readonly SymbolTableEnumerator _symbolTables;
+                
+        // In priority order. 
+        public SymbolTableResolver(SymbolTableEnumerator symbolTables)
+        {
+            _symbolTables = symbolTables;
+        }
+
+        // $$$ Needed?
+        IExternalDocument INameResolver.Document => default;
+
+        IExternalEntityScope INameResolver.EntityScope => throw new NotImplementedException();
+
+        DName INameResolver.CurrentProperty => default;
+
+        DPath INameResolver.CurrentEntityPath => default;
+
+        // Expose the list to aide in intellisense suggestions. 
+        public IEnumerable<TexlFunction> Functions
+        {
+            get
+            {
+                foreach (INameResolver table in _symbolTables)
+                {
+                    foreach (var function in table.Functions)
+                    {
+                        yield return function;
+                    }
+                }
+            }
+        }
+
+        IExternalEntity INameResolver.CurrentEntity => null;
+
+        public bool SuggestUnqualifiedEnums => false;
+
+        public IReadOnlyDictionary<string, NameLookupInfo> GlobalSymbols
+        {
+            get
+            {
+                var map = new Dictionary<string, NameLookupInfo>();
+
+                foreach (var table in _symbolTables)
+                {
+                    if (table is IGlobalSymbolNameResolver x)
+                    {
+                        // $$$ Merge. 
+                        // Tough to do with an IDictionary. Change to IEnumerable.
+                        //return x.GlobalSymbols;
+                        foreach (var kv in x.GlobalSymbols)
+                        {
+                            if (!map.ContainsKey(kv.Key))
+                            {
+                                map[kv.Key] = kv.Value;
+                            }
+                        }
+                    }
+                }
+
+                //return new Dictionary<string, NameLookupInfo>();  // empty
+                return map;
+            }
+        }
+
+        // $$$ MAke this non-virtual 
+        public virtual bool Lookup(DName name, out NameLookupInfo nameInfo, NameLookupPreferences preferences = NameLookupPreferences.None)
+        {
+            foreach (INameResolver table in _symbolTables)
+            {
+                if (table.Lookup(name, out nameInfo, preferences))
+                {
+                    return true;
+                }
+            }
```

Comment:
Is Linq that slow to rewrite it here?

Reply:
?
The use of out-parameters discourages me from trying to do this in a linq query.